examples_tests/43.AllocatorTest/main.cpp

```diff
@@ -1,53 +1,219 @@
 #define _IRR_STATIC_LIB_
-
 #include <irrlicht.h>
-#include "irr/core/alloc/GeneralpurposeAddressAllocator.h"
-
-using namespace irr;
+#include <random>
+#include <cmath>
 
+using namespace irr;
+using namespace core;
 
-/*
-	Problems with GeneralPurposeallocator:
-		1. GeneralPurposeallocator is not able to allocate as much address space as it is expected to
-		2. Program doesn't compile, when I try to call safe_shrink_size
-		3. "Assertion failed: false, file *\include\irr\core\alloc\GeneralpurposeAddressAllocator.h, line 368"
+/*
+	If traits say address allocator supports arbitrary order free,
+	then ETAT_CHOOSE_RANDOMLY is valid. Otherwise ETAT_CHOOSE_MOST_RECENTLY_ALLOCATED
```

Comment:
stack allocator is "most recently allocated" but the linear allocator has `free_addr` as a NO-OP, basically the whole allocator needs to be `reset` to actually reclaim the memory.